packages-common/pluggable-widgets-tools/utils/native/common.ts

```diff
@@ -0,0 +1,25 @@
+import { ImageStyle, TextStyle, ViewStyle } from "react-native";
+
+interface CustomStyle {
+    [key: string]: string | number;
+}
+
+export interface Style {
+    [key: string]: CustomStyle | ViewStyle | TextStyle | ImageStyle;
+}
+
+export function flattenStyles<T extends Style>(defaultStyle: T, overrideStyles: Array<T | undefined>): T {
```

Comment:
Maybe it is the time to give it a more public name? Like `mergeNativeStyles` or `mergeStyles`? You can also add short documentation comment, since afaik it will be copied to output by tsc. /cc @diego-antonelli

Reply:
`mergeNativeStyles` looks better, for the web, `parseInlineStyles`

Reply:
Renamed it